managed/ui/src/components/config/Storage/StorageConfiguration.js

```diff
@@ -105,19 +106,49 @@ class StorageConfiguration extends Component {
   addStorageConfig = (values, action, props) => {
     const type =
       (props.activeTab && props.activeTab.toUpperCase()) || Object.keys(storageConfigTypes)[0];
-    Object.keys(values).forEach((key) => {
+      Object.keys(values).forEach((key) => {
       if (typeof values[key] === 'string' || values[key] instanceof String)
         values[key] = values[key].trim();
     });
     const dataPayload = { ...values };
-    if (props.activeTab === 's3') {
-      if (values['IAM_INSTANCE_PROFILE']) {
+
+    // These conditions will remove all the non-required JSON keys from
+    // the respective tab.
+    switch (props.activeTab) {
+      case 'nfs':
         delete dataPayload['AWS_ACCESS_KEY_ID'];
         delete dataPayload['AWS_SECRET_ACCESS_KEY'];
-      }
-      if ('IAM_INSTANCE_PROFILE' in dataPayload) {
-        dataPayload['IAM_INSTANCE_PROFILE'] = dataPayload['IAM_INSTANCE_PROFILE'].toString();
-      }
+        delete dataPayload['AWS_HOST_BASE'];
+        delete dataPayload['AZURE_STORAGE_SAS_TOKEN'];
+        delete dataPayload['GCS_CREDENTIALS_JSON'];
+        break;
+
+      case 'gcs':
+        delete dataPayload['AWS_ACCESS_KEY_ID'];
+        delete dataPayload['AWS_SECRET_ACCESS_KEY'];
+        delete dataPayload['AWS_HOST_BASE'];
+        delete dataPayload['AZURE_STORAGE_SAS_TOKEN'];
+        break;
+
+      case 'az':
+        delete dataPayload['AWS_ACCESS_KEY_ID'];
+        delete dataPayload['AWS_SECRET_ACCESS_KEY'];
+        delete dataPayload['AWS_HOST_BASE'];
+        delete dataPayload['GCS_CREDENTIALS_JSON'];
```

Comment:
it's not clear what props come into the final payload, so I'd suggest using `_.pick()` instead of multiple deletes
that way code would be shorter and have a clear indication of what props lands into the payload

Reply:
Done.